src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.html

```diff
@@ -22,18 +22,32 @@
     </span>
   </div>
   <div class="card-body fill-plain">
-    <div>
+    <form #courseForm="ngForm" (ngSubmit)="onSaveCourse()">
       <div class="form-row form-group">
         <label class="col-sm-3">Course ID:</label>
         <div class="col-sm-3">
-          <input class="col-xs-12 form-control" maxlength="40" tabindex="1" [(ngModel)]="course.courseId" disabled>
+          <input class="col-xs-12 form-control" type="text" tabindex="1"
+                 #courseId="ngModel" [(ngModel)]="course.courseId" name="courseId"
+                 [maxlength]="FormValidator.COURSE_ID_MAX_LENGTH" required disabled>
+          <div [hidden]="courseId.valid || (courseId.pristine && courseId.untouched)"
+               class="invalid-field">
+            <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
+            The field Course ID should not be empty.
+          </div>
         </div>
       </div>
       <div class="form-row form-group">
         <label class="col-sm-3">Course Name:</label>
         <div class="col-sm-9">
-          <input class="form-control" ngbTooltip="The name of the course, e.g. Software Engineering."
-                 maxlength="64" tabindex="2" [(ngModel)]="course.courseName" [disabled]="!isEditingCourse">
+          <input class="form-control" ngbTooltip="Enter the name of the course, e.g. Software Engineering."
+                 type="text" tabindex="2" [disabled]="!isEditingCourse" name="courseName"
+                 [(ngModel)]="course.courseName" #courseName="ngModel"
+                 [maxlength]="FormValidator.COURSE_NAME_MAX_LENGTH" required>
```

Comment:
`instructor-course-student-edit-page.component.ts` does not use `maxlength ` on `input` component. It displays an extra exceed max length message. But having a `maxlength` on `input` also has its merit. Maybe we should discuss and standardise it?

Reply:
I was just hoping to replace the magic number there haha, I also realised `instructor-course-student-edit-page` was using a reactive form but the one being addressed by this pull request is a template-driven form which makes it more complicated maybe

But yes, could consider a "form migration" discussion down the road